test/funnels/automation/VaultMinter.t.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.16;
+
+import "dss-test/DssTest.sol";
+import { VaultMinter } from "src/funnels/automation/VaultMinter.sol";
+import { AllocatorRoles } from "src/AllocatorRoles.sol";
+import { AllocatorVault } from "src/AllocatorVault.sol";
+import { AllocatorBuffer } from "src/AllocatorBuffer.sol";
+import { VatMock } from "test/mocks/VatMock.sol";
+import { JugMock } from "test/mocks/JugMock.sol";
+import { GemMock } from "test/mocks/GemMock.sol";
+import { NstJoinMock } from "test/mocks/NstJoinMock.sol";
+
+contract VaultMinterTest is DssTest {
+    using stdStorage for StdStorage;
+
+    event Kiss(address indexed usr);
+    event Diss(address indexed usr);
+    event SetConfig(int64 num, uint32 hop, uint128 lot);
+    event Mint(uint128 lot);
+    event Burn(uint128 lot);
+
+    VatMock         public vat;
+    JugMock         public jug;
+    GemMock         public nst;
+    NstJoinMock     public nstJoin;
+    AllocatorBuffer public buffer;
+    AllocatorRoles  public roles;
+    AllocatorVault  public vault;
+    VaultMinter     public minter;
+
+    bytes32 constant ILK         = "aaa";
+    address constant FACILITATOR = address(0x1337);
+    address constant KEEPER      = address(0xb0b);
+    uint8   constant MINTER_ROLE  = uint8(1);
+
+    function setUp() public {
+        vat     = new VatMock();
+        jug     = new JugMock(vat);
+        nst     = new GemMock(0);
+        nstJoin = new NstJoinMock(vat, nst);
+        buffer  = new AllocatorBuffer();
+        roles   = new AllocatorRoles();
+        vault   = new AllocatorVault(address(roles), address(buffer), ILK, address(nstJoin));
+        vault.file("jug", address(jug));
+        buffer.approve(address(nst), address(vault), type(uint256).max);
+
+        vat.slip(ILK, address(vault), int256(1_000_000 * WAD));
+        vat.grab(ILK, address(vault), address(vault), address(0), int256(1_000_000 * WAD), 0);
+
+        minter = new VaultMinter(address(vault));
+
+        // Allow minter to perform operations in the vault
+        roles.setIlkAdmin(ILK, address(this));
+        roles.setRoleAction(ILK, MINTER_ROLE, address(vault), AllocatorVault.draw.selector, true);
+        roles.setRoleAction(ILK, MINTER_ROLE, address(vault), AllocatorVault.wipe.selector, true);
+        roles.setUserRole(ILK, address(minter), MINTER_ROLE, true);
+
+        // Set up keeper to mint and burn
+        minter.rely(FACILITATOR);
+        vm.startPrank(FACILITATOR);
+        minter.kiss(KEEPER);
+        vm.stopPrank();
+
+        vm.warp(1 hours);
+    }
+
+    function testConstructor() public {
+        VaultMinter m = new VaultMinter(address(0xABC));
```

Comment:
Can also check the existence of the `Rely` event here.

Reply:
Good catch, this is something we need to do in other contracts as well.

Reply:
Done.

Reply:
For the other contracts we should do a separate PR.